### DISCUSSION TITLE
Expand price rules

### DISCUSSION THREAD

**Discussion by torrfura** [swellstores/community#275] [Help] [answered]
How can I expand pricing rules in products list query?

```
await swell.products.list({
   expand: ['price']
});
```

The link / section exist, but no content. https://developers.swell.is/frontend-api/products#list-products-with-prices

**Comment by ericingram** [comment]
The price rule field is `prices` (not `price`), but since it's not a link field, you should put it in `fields: ['prices']`

**Comment by torrfura** [comment]
Maybe I'm not sure if I'm looking in the right direction here. I'm trying to offer retail pricing with price latters, and display it in a list of all products. Is it so that the price rules are not avaliable in this call?

```
await swell.products.list({
    fields: ['prices']
});
```
This is how I've set it up:
<img width="1202" alt="image" src="https://github.com/swellstores/community/assets/155930/c9514a92-b1ea-4bb9-98b5-04c3b88807c0">

this is my response, logged in with a customer that has the group "Retail":
```
{
    ...
    "price": 85,
    "purchase_options": {
        "standard": {
            "price": 85,
            "sale": false,
            "sale_price": null
        }
    },
    ...
}

**Reply by ericingram** [reply, marked as answer]
Storefront pricing logic will remove group-based price rules from the response unless there is an account logged in using that group. You can bypass it if you know the group ID in your frontend, like this:

```
await swell.products.list({
  $pricing: { account_group: "retail" }
})
```

**Reply by amaidah** [reply]
Is there a way to add the product to cart and checkout with a particular `account_group` pricing without the user being logged in? This would help for making price lists for sites where there is no requirement to be logged in

**Reply by ericingram** [reply]
Have you tried it with "Customer login: Optional" in checkout settings?

An account doesn't need to have a password, in which case it won't require login. I think if the account is set on the cart via API while it does not have a password defined, it will apply group pricing still.

**Reply by amaidah** [reply]
Yup, I have this set to `optional`:  "Customer login: Optional" and what you suggested, setting an email address without password via API does work to get the grouped pricing!

However, then you have a fake email being applied to a customer's checkout to get the group pricing. Is there any way to get group level pricing in cart without needing to use a fake email and without having a user signed in?

![image](https://github.com/swellstores/community/assets/16659483/f5ae2378-df3e-4cc7-acf7-ff3e76a97ca1)

**Reply by ericingram** [reply]
I don't believe there is a way to do it using the Frontend API, but it can be done using the Backend API by passing `$pricing: { account_group }`.

Once Swell's App framework gets to production later this year, you'll be able to make serverless functions that can handle this kind of custom logic called from the frontend. Until then, the frontend API is limited to certain default functionality as such.

**Comment by torrfura** [comment]
What do you think? Is it just me missing something obvious?

**Comment by torrfura** [comment]
Perfect! Even though I was logged in as user retail (custom storefront) it didn't work, but that $pricing key helped me!

Thank you!
